content/docs/rfcs/51/README.md

```diff
@@ -173,45 +169,103 @@ This example node is part of shards `13`, `14`, and `45` in the Status main-net
+
+### Example
+
+- Full length `/0/unbiased/myapp/1/mytopic/cbor`
+- Short length `/myapp/1/mytopic/cbor`
+
+### Generation
+The generation number monotonously increases and indirectly refers to the total number of shards of the Waku Network.
+
+The first generation (zero) MUST use 8 shards in total.
+The cluster index is 1 and the indices of each shards are numbered 0 to 7.
+
+This document will be updated for future generations.
+
+### Bias
+Bias is used to skew the priority of shards via weights.
+Other biases than `unbiased` are unspecified for now but may be used in the future.
+
+### Topic Design
+Content topics have 2 purposes: filtering and routing.
+Filtering is done by changing the `{content-topic-name}` field.
+As this part is not hashed, it will not affect routing (shard selection).
+The `{application-name}` and `{version-of-the-application}` fields do affect routing.
+Using multiple content topics with different `{application-name}` field has advantages and disadvantages.
+It increases the traffic a relay node is subjected to when subscribed to all topics.
+It also allows relay and light nodes to subscribe to a subset of all topics.
+
+## Problems
+
+### Hot Spots
+
+Hot spots occur (similar to DHTs), when a specific mesh network (shard) becomes responsible for (several) large multicast groups (content topics).
+The opposite problem occurs when a mesh only carries multicast groups with very few participants: this might cause bad connectivity within the mesh.
 
-There are (at least) two issues that have to be solved: *Hot spots* and *Discovery* (see next subsection).
+The current autosharding method cannot solve this problem.
 
-Hot spots occur (similar to DHTs), when a specific mesh network becomes responsible for (several) large multicast groups (content topics).
-The opposite problem occurs when a mesh only carries multicast groups with very few participants: this might cause bad connectivity within the mesh.
-Our research goal here is finding efficient ways of distribution.
-We could get inspired by the DHT literature.
-We also have to consider:
-If a node is part of many content topics which are all spread over different shards,
-the node will potentially be exposed to a lot of network traffic.
+A new version of autosharding based on network traffic measurements could be designed but would require further research and development.
 
-## Discovery
+### Discovery
 
 For the discovery of automatic shards this document specifies two methods (the second method will be detailed in a future version of this document).
 
 The first method uses the discovery introduced above in the context of static shards.
-The index range `49152 - 65535` is reserved for automatic sharding.
-Each index can be seen as a hash bucket.
-Consistent hashing maps content topics in one of these buckets.
+Each cluster index can be seen as a hash bucket.
+Rendezvous hashing maps content topics in one of these buckets.
```

Comment:
I thought rendezvous hashing was mapping content topics to shards.
Can you clarify how each cluster is hash bucket?

Reply:
I only changed consistent for rendezvous hashing. What he meant IDK.

Maybe remove this as it's confusing?